# Predicting the quality of wine using pH (Tensorflow)

In [22]:
import tensorflow as tf
from tensorflow import keras
import numpy as np
import pandas as pd

In [23]:
df = pd.read_csv("E:/Machine learning/datasets/winequality-red.csv")

In [24]:
df

fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0               7.4             0.700         0.00             1.9      0.076   
1               7.8             0.880         0.00             2.6      0.098   
2               7.8             0.760         0.04             2.3      0.092   
3              11.2             0.280         0.56             1.9      0.075   
4               7.4             0.700         0.00             1.9      0.076   
...             ...               ...          ...             ...        ...   
1594            6.2             0.600         0.08             2.0      0.090   
1595            5.9             0.550         0.10             2.2      0.062   
1596            6.3             0.510         0.13             2.3      0.076   
1597            5.9             0.645         0.12             2.0      0.075   
1598            6.0             0.310         0.47             3.6      0.067   

      free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                    11.0                  34.0  0.99780  3.51       0.56   
1                    25.0                  67.0  0.99680  3.20       0.68   
2                    15.0                  54.0  0.99700  3.26       0.65   
3                    17.0                  60.0  0.99800  3.16       0.58   
4                    11.0                  34.0  0.99780  3.51       0.56   
...                   ...                   ...      ...   ...        ...   
1594                 32.0                  44.0  0.99490  3.45       0.58   
1595                 39.0                  51.0  0.99512  3.52       0.76   
1596                 29.0                  40.0  0.99574  3.42       0.75   
1597                 32.0                  44.0  0.99547  3.57       0.71   
1598                 18.0                  42.0  0.99549  3.39       0.66   

      alcohol  quality  
0         9.4        5  
1         9.8        5  
2         9.8        5  
3         9.8        6  
4         9.4        5  
...       ...      ...  
1594     10.5        5  
1595     11.2        6  
1596     11.0        6  
1597     10.2        5  
1598     11.0        6  

[1599 rows x 12 columns]

In [34]:
l1 = df['pH'].tolist()
l2 = df['quality'].tolist()

In [35]:
model = tf.keras.Sequential([keras.layers.Dense(units=1, input_shape=[1])])

In [36]:
model.compile(optimizer="sgd", loss="mean_squared_error")

In [37]:
x = np.array(l1, dtype=float)
y = np.array(l2, dtype=float)

In [38]:
model.fit(x,y,epochs=500)

Epoch 1/500
50/50 [==============================] - 0s 1ms/step - loss: 0.9591
Epoch 2/500
50/50 [==============================] - 0s 2ms/step - loss: 0.7445
Epoch 3/500
50/50 [==============================] - 0s 2ms/step - loss: 0.7435
Epoch 4/500
50/50 [==============================] - 0s 2ms/step - loss: 0.7415
Epoch 5/500
50/50 [==============================] - 0s 1ms/step - loss: 0.7443
Epoch 6/500
50/50 [==============================] - 0s 1ms/step - loss: 0.7415
Epoch 7/500
50/50 [==============================] - 0s 2ms/step - loss: 0.7414
Epoch 8/500
50/50 [==============================] - 0s 3ms/step - loss: 0.7395
Epoch 9/500
50/50 [==============================] - 0s 2ms/step - loss: 0.7424
Epoch 10/500
50/50 [==============================] - 0s 2ms/step - loss: 0.7396
Epoch 11/500
50/50 [==============================] - 0s 1ms/step - loss: 0.7396
Epoch 12/500
50/50 [==============================] - 0s 1ms/step - loss: 0.7392
Epoch 13/500
50/50 [=================

In [41]:
print(model.predict([3.39]))

1/1 [==============================] - 0s 31ms/step
[[5.591528]]


# The value is 5.591528 which is very close to 6.